README.md

```diff
@@ -20,9 +18,127 @@ it, simply add the following line to your Podfile:
 pod 'FirebaseHelper'
 ```
 
-## Author
+## Usage
+
+Initialize an instance of `FirebaseHelper`:
+
+```swift
+import Firebase
+import FirebaseHelper
+
+let firebaseHelper = FirebaseHelper(FirebaseDatabase.Database.database().reference())
+```
+
+`FirebaseHelper(_ ref: DatabaseReference)` takes in a `DatabaseReference`. Generally you'd want this to be the root of your database.
+
+For convenience, you can put this in say, a common utils file, in your project:
+
+```swift
+extension FirebaseHelper {
+    static let main: FirebaseHelper = {
+        FirebaseHelper(FirebaseDatabase.Database.database().reference())
+    }()
+}
+```
+
+Now, anywhere in your project, you can simply call `FirebaseHelper.main` to access this instance of `FirebaseHelper`.
+
+### Common Database Operations
+
+#### Get
+
+Example:
+
+```swift
+FirebaseHelper.main.get("users", "john123", "favoriteFood") { result in
+    switch result {
+      case .failure(let error):
+        // handle error
+      case .success(let data):
+        // get string from data
+    }
+}
+```
+
+API:
+
+```swift
+public func get(_ first: String, _ rest: String..., completion: @escaping (Result<DataSnapshot, Error>) -> Void)
+```
+
+`get()` takes in a variadic parameter of child nodes that will be built on the `DatabaseReference` you initialized your instance of `FirebaseHelper` on.
+
+The callback returns a `Result`:
+
+```swift
+public enum Result<T, Error> {
+    case success(T)
+    case failure(Error)
+}
+```
+
+In this case, `T` is `DataSnapshot`. An error case will either be because an invalid child was passed in or some other error in your database.
+
+#### Set, Delete, Increment
+
+`set()`, `delete()`, and `increment()` work similarly, but instead of returning a `Result`, they simply return an `Error` if one occurred, or `nil` otherwise.
+
+Examples:
+
+```swift
+// The first parameter is an `Any` that gets set at the specified path.
+FirebaseHelper.main.set("pizza", at: "users", "john123", "favoriteFood") { error in
+    if let error = error {
+      // handle error
+  }
+}
+
+FirebaseHelper.main.delete("users", "john123", "favoriteFood") { error in
+    if let error = error {
+      // handle error
+  }
+}
+
+FirebaseHelper.main.increment(5, "users", "john123", "favoriteFoodEatenThisMonth") {
```

Comment:
This function confuses me:

Is 5 the value favoriteFoodEatenThisMonth is being "increment"ed to? Or will the value be "increment"ed to 6 (5++)? Also why is this necessary? (how is this different from `.set`)

Reply:
This is confusing. It will increment _by_ 5. I'll add that parameter name for clarity.

[Firebase docs](https://firebase.google.com/docs/database/ios/read-and-write#save_data_as_transactions)

This is necessary because one cannot simply say `set(oldValue + 5)` because of race conditions. Say concurrent `set()`s are taking place, `oldValue`s can go stale before `oldValue + 5`s take place, etc. Thus this API is a wrapper around Firebase's solution for this problem.

Reply:
ahhh okay that makes sense. Now that I think about it, thats actually extremely useful! Thanks for the explanation.